src/lib/core/PeerId.h

```diff
@@ -0,0 +1,59 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+namespace chip {
+
+/// Convenience typess to make it clear what different number types mean
+using NodeId   = uint64_t;
+using FabricId = uint64_t;
+
+constexpr NodeId kUndefinedNodeId = 0ULL;
+constexpr NodeId kAnyNodeId       = 0xFFFFFFFFFFFFFFFFULL;
+
+constexpr FabricId kUndefinedFabricId = 0ULL;
+
+/// A peer is identified by a node id within a fabric
+class PeerId
+{
+public:
+    PeerId() {}
+
+    NodeId GetNodeId() const { return mNodeId; }
```

Comment:
But why?

This seems like some Java inspired "everything must have a setter" code but this is not idiomatic in C++.

I'd request that we just do:

```
struct PeerId {
    NodeId nodeId     = kUndefinedNodeId;
    FabricId fabricId = kUndefinedFabricId;
};

// out of line operator==

```

Reply:
I wanted to avoid calls of `Foo(nodeid, fabricid)` since I often switched the order (logically fabric seems higher scope, so we want fabric, node order, but often we conside 'node' the primary and it goes first and that drives me insane.

Having `PeerId(1,2)` is no better, so I wanted to see code like `PeerId().SetFabric(1).SetNode(2)` or `PeerId().SetNode(1).SetFabric(2)` which is obvious.

I can also make a stuct, but that would require me building the struct separately or someone at some point *will* add the `PeerId(int64, int64)` constructor as a convenience and that will drive me insane again :( Happy to refactor on any suggestions though.

Reply:
We could also make NodeId/FabricId an actual one-member struct instead of a typedef ... but that will result in a very large refactor.